packages/db/postgres/migrations/001-user-used-storage.sql

```diff
@@ -0,0 +1,36 @@
+CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT);
+
+DROP FUNCTION user_used_storage(bigint);
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
+RETURNS used_storage
+LANGUAGE plpgsql
+AS
+$$
+DECLARE used_storage used_storage;
+BEGIN
+  SELECT COALESCE(SUM(c.dag_size), 0)
+  INTO used_storage.uploaded::TEXT
+  FROM upload u
+  JOIN content c ON c.cid = u.content_cid
+  WHERE u.user_id = query_user_id::BIGINT
+  AND u.deleted_at is null;
+
+  -- NOTE: auth_key.deleted_at is NOT included in the where clause, since the 
+  -- auth_key is only used to authorize a request.
+  SELECT COALESCE((
+    SELECT COALESCE(SUM(c.dag_size), 0)
+    FROM psa_pin_request psa_pr
+    JOIN content c ON c.cid = psa_pr.content_cid
+    JOIN pin p ON p.content_cid = psa_pr.content_cid
+    JOIN auth_key a ON a.id = psa_pr.auth_key_id
+    WHERE a.user_id = query_user_id::BIGINT
+    AND psa_pr.deleted_at is null
+    AND p.status = 'Pinned'
+    GROUP BY p.status
```

Comment:
We need to pick out distinct CIDs here before aggregating the `dag_size`. We only want to sum the `dag_size` for unique root CIDs stored for a given user, rather than every redundant copy of that data (pin) that we store.

In prod, there will generally be 3 or more pin rows for a given `content_cid`; one for each cluster node the data is duplicated to. 

I've see 15(!?) rows on `bafybeidluj5ub7okodgg5v6l4x3nytpivvcouuxgzuioa6vodg3xt2uqle` that has dag_size: `140147` for my user id.
<img width="979" alt="Screenshot 2022-03-24 at 14 21 20" src="https://user-images.githubusercontent.com/58871/159937448-532181db-77cf-4a36-869b-1d056f433a71.png">

Add the sum on dag_size is coming back as `2102205` rather than `140147`

<img width="880" alt="Screenshot 2022-03-24 at 14 21 49" src="https://user-images.githubusercontent.com/58871/159938665-b4355c22-a8ba-4426-a01d-be887d37b514.png">

I need to look in to why we have 15 rows and not ~3 for that CID, but that's a seperate issue.

Reply:
Corrected to aggregate the size for distinct CIDs:

![Screenshot 2022-03-25 at 17 08 16](https://user-images.githubusercontent.com/671408/160168445-88d4119d-e53c-4ae0-b91b-bd2435a60fc4.png)